src/mpm.ts

```diff
@@ -35,21 +36,31 @@ export async function setup(platform: string, architecture: string): Promise<str
     return mpm
 }
 
-export async function install(mpmPath: string, release: string, products: string[], destination: string) {
+export async function install(mpmPath: string, release: string, products: string[], destination: matlab.ToolcacheLocation) {
+    if (destination.useExisting) {
```

Comment:
Seems strange that a destination/location has a `useExisting` field. Maybe these are 2 separate things? What if `toolcacheLocation` just returned a tuple (e.g. `[location, existed] = toolcacheLocation`) and then there is logic to decide whether install should be called or not?

Reply:
That's a good point. Updated